### PR TITLE
#111 #115 Eliminate custom formatters in logging.py

### PR DESCRIPTION
**Pull request by kalbfled**
I deleted 2 custom formatters in logging.py that were not implemented correctly and are not necessary in the first place.  The abundant "failed to format log message" errors in our CloudWatch logs, which motivated the underlying tickets, were getting raised in these deleted formatters.  I also added new unit tests.

I opened [notification-api#1222](https://github.com/department-of-veterans-affairs/notification-api/issues/1222) to test these changes.  I deployed that branch to Dev, sent myself an SMS, and Cris ran her regression suite against Dev.  Neither of these actions produced error message in CloudWatch.

#111
#115

### CODE REVIEW

**Review by cris-oddball — APPROVED** [review]

File: notifications_utils/logging.py [hunk 1]
```diff
@@ -51,24 +51,18 @@ def after_request(response):
...
             'url': request.url,
-            'status': response.status_code
+            'status': response.status_code,
         }
 
         if 'service_id' in g:
-            extra_fields.update({
-                'service_id': g.service_id
-            })
+            extra_fields['service_id'] = g.service_id
```
Comment (k-macmillan) [inline notifications_utils/logging.py:58]: Are we certain this will not be a list? `.update()` gives us added options that are being removed.

**Review by k-macmillan — CHANGES_REQUESTED** [review]

File: notifications_utils/logging.py (hunk 1 above)
Comment (kalbfled) [reply notifications_utils/logging.py:58]: Yes.  "extra_fields" is declared as a dictionary right above this block.

Comment (k-macmillan) [reply notifications_utils/logging.py:58]: Not quite what I meant. The [update method](The specified items can be a dictionary, or an iterable object with key value pairs.) gives us the option of passing a dictionary (or other options). Are we certain we are okay losing that functionality?

File: notifications_utils/logging.py [hunk 2]
```diff
@@ -101,57 +95,44 @@ def ensure_log_path_exists(path):
...
-    handler.setFormatter(formatter)
-    handler.addFilter(AppNameFilter(app.config['NOTIFY_APP_NAME']))
-    handler.addFilter(RequestIdFilter())
+    if app.debug:
+        # Human readable stdout logs that omit static route 200 responses
+        logging.getLogger('werkzeug').addFilter(is_200_static_log)
+        stream_handler.setFormatter(logging.Formatter(LOG_FORMAT, TIME_FORMAT))
+    else:
+        # Machine readable json to stdout
+        stream_handler.setFormatter(JsonFormatter(LOG_FORMAT, TIME_FORMAT))
```
Comment (k-macmillan) [inline notifications_utils/logging.py:125]: Why logging formatter if in debug but json formatter if not in debug?

File: notifications_utils/logging.py (hunk 1 above)
Comment (kalbfled) [reply notifications_utils/logging.py:58]: Yes.  "g" is the [Flask global context](https://flask.palletsprojects.com/en/2.3.x/api/#flask.g), and `g.service_id` is a primitive type.
```
...va/notification-api$ grep -rn -e "\bg\.service_id" .
./app/authentication/auth.py:66:        g.service_id = current_app.config.get('ADMIN_CLIENT_USER_NAME')
./app/authentication/auth.py:179:        g.service_id = api_key.service_id
```

File: notifications_utils/logging.py [hunk 3]
```diff
@@ -160,46 +141,9 @@ class RequestIdFilter(logging.Filter):
...
         if has_request_context() and hasattr(request, 'request_id'):
             return request.request_id
-        else:
-            return 'no-request-id'
 
-    def filter(self, record):
-        record.request_id = self.request_id
-
-        return record
+        return 'no-request-id'
```
Comment (k-macmillan) [inline notifications_utils/logging.py:145]: return request.request_id if has_request_context() and hasattr(request, 'request_id') else 'no-request-id'

File: setup.py [hunk 4]
```diff
@@ -36,7 +37,7 @@
         'orderedset>=2.0.3',
         'phonenumbers~=8.12.12',
         'pypdf2 ~= 3.0.1',
-        'python-json-logger>=0.1.11',
+        'python-json-logger~=2.0.7',
```
Comment (k-macmillan) [inline setup.py:40]: WOW, what an update.

File: notifications_utils/logging.py (hunk 2 above)
Comment (kalbfled) [reply notifications_utils/logging.py:125]: I preserved the existing behavior.  I think the intent is not to clutter your shell when running locally and to use a machine-parsable format with CloudWatch.

**Review by k-macmillan — CHANGES_REQUESTED** [review]

File: setup.py (hunk 4 above)
Comment (kalbfled) [reply setup.py:40]: Note the "greater than."  I just made it match what is already in use.

File: notifications_utils/logging.py (hunk 1 above)
Comment (k-macmillan) [reply notifications_utils/logging.py:58]: Does `g.endpoint` behave the same?

File: setup.py (hunk 4 above)
Comment (k-macmillan) [reply setup.py:40]: It was just shocking that it hadn't been coded to anything newer.

File: notifications_utils/logging.py (hunk 3 above)
Comment (kalbfled) [reply notifications_utils/logging.py:145]: I dig it, but that will give me a Flask8 error about the line being too long.  Then I would have to back-slash it, and that less readable that what's already there.

File: notifications_utils/logging.py (hunk 1 above)
Comment (kalbfled) [reply notifications_utils/logging.py:58]: Yes.  This should just be a URL string.
```
...va/notification-api$ grep -rn -e "\bg\.endpoint" .
./app/__init__.py:406:        g.endpoint = request.endpoint
```

Comment (k-macmillan) [reply notifications_utils/logging.py:58]: Great, thanks!

File: notifications_utils/logging.py (hunk 3 above)
Comment (k-macmillan) [reply notifications_utils/logging.py:145]: Isn't flake8 120 columns?

Comment (k-macmillan) [reply notifications_utils/logging.py:145]: Yeah, found it:
image

Comment (kalbfled) [reply notifications_utils/logging.py:145]: That sounds correct, but this is 124 columns with the preceding indentation.

**Review by k-macmillan — APPROVED** [review]

**Pull request merged**